components/uaepass-authenticator/src/main/java/org/wso2/carbon/identity/uaepass/authenticator/UAEPassAuthenticator.java

```diff
@@ -0,0 +1,1109 @@
+                return UAEPassAuthenticatorConstants.Endpoints.ProductionEndpointValues.UAEPASS_PROD_LOGOUT_ENDPOINT_VALUE;
+            } else {
+                return getAuthenticatorConfig().getParameterMap().get
+                        (UAEPassAuthenticatorConstants.Endpoints.ProductionEndpointKeys.UAEPASS_PROD_LOGOUT_ENDPOINT_KEY);
+            }
+        }
+    }
+
+
+    /**
+     * Allow to select the UAEPass Env to be used. By default, env would pick according to the client id.
+     *
+     * @param context
+     * @return
+     * @throws IOException
+     */
+    public String getUAEPassEnvironment(AuthenticationContext context) {
+
+        Map<String, String> authenticatorProperties = context.getAuthenticatorProperties();
+        if (isStagingEnvSelected(context)) {
+                return UAEPassAuthenticatorConstants.UAEPassRuntimeConstants.STAGING;
+        } else {
+            if (StringUtils.equals(authenticatorProperties.get(UAEPassAuthenticatorConstants.UAE.CLIENT_ID),
+                    UAEPassAuthenticatorConstants.UAEPassRuntimeConstants.SANDBOX_STAGE_CLIENT_ID)) {
+                return UAEPassAuthenticatorConstants.UAEPassRuntimeConstants.STAGING;
+            } else {
+                return UAEPassAuthenticatorConstants.UAEPassRuntimeConstants.PRODUCTION;
+            }
+        }
+    }
+
+
+    /**
+     * Map the JSON payload attributes as key value pairs
+     *
+     * @param jwtAttributeSet
+     * @return
+     */
+    public Map<String,Object> buildJSON(Set<Map.Entry<String, Object>> jwtAttributeSet){
+
+        Map<String, Object> jwtAttributeMap = new HashMap<String,Object>();
+        for (Map.Entry<String, Object> entry : jwtAttributeSet) {
+            jwtAttributeMap.put(entry.getKey(), entry.getValue());
+            if(LOG.isDebugEnabled()){
+                LOG.debug("UAEPass user claim : " +entry.getKey());
+            }
+        }
+        return jwtAttributeMap;
+    }
+
+
+    /**
+     *
+     *
+     * @param context
+     * @return
+     */
+    public boolean isStagingEnvSelected(AuthenticationContext context){
+        Map<String, String> authenticatorProperties = context.getAuthenticatorProperties();
+        boolean isStagingEnv = Boolean.parseBoolean(authenticatorProperties.get(UAEPassAuthenticatorConstants.
+                UAE.UAEPASS_ENV));
+        if(isStagingEnv){
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+
+    /**
+     *
+     * @param context
+     * @return
+     */
+    public boolean isLogoutEnabled(AuthenticationContext context) {
+        Map<String, String> authenticatorProperties = context.getAuthenticatorProperties();
+        boolean isLogoutEnabled = Boolean.parseBoolean(authenticatorProperties
+                .get(UAEPassAuthenticatorConstants.UAE.LOGOUT_ENABLE));
+        if(isLogoutEnabled){
+            return true;
+        } else {
+            return false;
+        }
+    }
+}
```

Comment:
Newline at the end of the file.
```suggestion
}

```

Reply:
Fixed with : [bf5488f](https://github.com/wso2-extensions/identity-outbound-auth-uaepass/pull/1/commits/bf5488f695a7cae8933ece11d28fcab07cd8e53f)